distributed/utils.py

```diff
@@ -386,7 +386,7 @@ def key_split_group(x):
     typ = type(x)
     if typ is tuple:
         return x[0]
-    elif typ is str:
+    elif typ is unicode or typ is str:
```

Comment:
OK, so it looks like a 5-20% slowdown.  This *could* go in, but if you're willing it might be nicer to build separate Python 2 and Python 3 versions of this function and decide which to use on import time.

```python
if PY3:
    def ...
        ...
else:
    def ...
```